test/framework/src/main/java/org/opensearch/test/ConcurrentSearchIntegTestCase.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.test;
+
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+import org.junit.Before;
+import org.opensearch.action.admin.cluster.settings.ClusterUpdateSettingsResponse;
+import org.opensearch.common.settings.FeatureFlagSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.FeatureFlags;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.opensearch.search.SearchService.CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING;
+
+public class ConcurrentSearchIntegTestCase extends OpenSearchIntegTestCase {
```

Comment:
I think we should name it as a generic class something like `DynamicSettingBasedFeatureIntegTestCase` or `DynamicSettingParameterizedIntegTestCase` such that it can be used by other features as needed. Then each test will provide the `parameters` function with set of settings to enable/disable. For example: in this case for concurrent search the tests identified and planned to be extended with this new base class will provide the `Settings` object that needs to be applied for enabling/disabling the feature. That way we can keep this mechanism generic. If for a test it needs to enable/disable multiple features it can do so by providing the settings object with different combinations of feature setting values. Something like below:

```
new Object[] { Settings.builder().put(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING.getKey(), true).put(Feature_1.getKey(), true).build() },
            new Object[] { Settings.builder().put(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING.getKey(), false).put(Feature_1.getKey(), false).build()}
....
....
```

Reply:
I'm not sure this is a good idea, but could we even build this into OpenSearchIntegTestCase? Something like:

```
private final Settings customClusterSettings;

public OpenSearchIntegTestCase() {
    this(Settings.EMPTY);
}

public OpenSearchIntegTestCase(Settings settings) {
    this.customClusterSettings = settings;
}
```

Tests that want to parameterize based on settings can do so by overriding that constructor, but otherwise can continue to use the default. (The `Settings` object may not be the right structure to pass in...maybe we need something that combines both settings and feature flags?).